### PR TITLE
Actions wiki deploy

### PR DESCRIPTION
**Pull request by maxb2**
This adds the contents of the wiki to the repository and adds an action to deploy any changes to the `wiki/` directory.

~~Blocked by #594.~~

@n1ru4l, can you set a new secret `GH_PAT` with the token generated from [here](https://github.com/settings/tokens/new?scopes=repo&description=wiki%20page%20creator%20token)? It's for [Decathlon/wiki-page-creator-action](https://github.com/marketplace/actions/wiki-page-creator-action)

### CODE REVIEW

File: .github/workflows/deploy-wiki.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      with: 
+        # Fetch merge commit and last commit from pull request
+        fetch-depth: 2
+
+    - name: Set environment
+      run: |
+        # Set the author name, author email, and message from the pull request
+        echo "::set-env name=ACTION_MAIL::$(git log -n 1 --skip=1 --pretty=format:%ae)"
+        echo "::set-env name=ACTION_NAME::$(git log -n 1 --skip=1 --pretty=format:%an)"
+        echo "::set-env name=WIKI_PUSH_MESSAGE::$(git log -n 1 --skip=1 --pretty=format:%s)"
```
Comment (maxb2) [inline .github/workflows/deploy-wiki.yml:27]: I'm not in love with this method of extracting the original author. When a wiki pull request gets merged into master you get the following commit history:
```
commit c1a7a1cf0e61da8c8fc8227ffed21daec6e8e046 (HEAD -> master)
Merge: 160a357 e1c7ada
Author: maxb2 <maxb2@mail.missouri.edu>
Date:   Wed Jul 15 10:21:57 2020 -0500

    Merge branch 'throwaway700-patch-6'

commit e1c7adaa0302df6a7d41a908945eedee7cff1f6d (throwaway700-patch-6)
Author: throwaway700 <68336138+throwaway700@users.noreply.github.com>
Date:   Wed Jul 15 10:17:52 2020 -0500

    wiki update
````

So, you have to fetch the last two commits and skip one of them to get the author info. I'm open to other ways of doing this.

File: .dockerignore [hunk 2]
```diff
@@ -8,3 +8,7 @@ docker-cache
 arm-compile.Dockerfile
 Dockerfile
 .github/
+docs/
+wiki/
+.vscode
+*.md
```
Comment (maxb2) [inline .dockerignore:14]: This excludes unnecessary files in the `COPY . .` stage of the docker build. Hopefully, when there is only a change to the docs the docker build for the `dev` tag will be completely cached.

**Comment by maxb2** [comment]
If we want to use PRs for wiki edits, we need to [change the access permissions for the wiki](https://docs.github.com/en/github/building-a-strong-community/changing-access-permissions-for-wikis). Currently it's open to anyone to edit the wiki. We should add a page on how to contribute to the wiki as well.

File: .github/workflows/deploy-wiki.yml (hunk 1 above)
Comment (n1ru4l) [reply .github/workflows/deploy-wiki.yml:27]: Will this work if the commit gets squashed into a single commit?

File: wiki/Hosting.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Under construction!
```
Comment (n1ru4l) [inline wiki/Hosting.md:1]: This is out of sync with the wiki, someone already added content. I also disabled wiki edits from non maintainers. We should add a Not into the sidebar/footer on how the contents can be edited.

File: .github/workflows/deploy-wiki.yml (hunk 1 above)
Comment (maxb2) [reply .github/workflows/deploy-wiki.yml:27]: I don't think so. Can you think of another way to extract the author and commit message from a pull request? If we don't do this, we would still have the correct history in the original repo, but the wiki will always say dungeon-revealer is the author and have no commit message.

Comment (maxb2) [reply .github/workflows/deploy-wiki.yml:27]: I could have it check if the latest commit author is dungeon-revealer, skip back one commit and check again if true, else get the author and message.

File: .github/workflows/deploy-wiki.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          echo "Skipping Merge commit"
+          CMD='git log -n 1 --skip=1'
+        else
+          echo "Using current commit"
+          CMD='git log -n 1'
+        fi
+        # Set the author name, author email, and message from the pull request
+        echo "::set-env name=ACTION_MAIL::$(eval $CMD --pretty=format:%ae)"
+        echo "::set-env name=ACTION_NAME::$(eval $CMD --pretty=format:%an)"
+        echo "::set-env name=WIKI_PUSH_MESSAGE::$(eval $CMD --pretty=format:%s)"
```
Comment (maxb2) [inline .github/workflows/deploy-wiki.yml:35]: @n1ru4l, This checks if HEAD is a merge commit and skips back one commit if it is. This should handle squashed commits the right way.

Comment (n1ru4l) [reply .github/workflows/deploy-wiki.yml:35]: It seems a bit hacky but should do the job, so let's keep it like that. We can come up with a better solution later 😊

File: wiki/Contributing.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+The files for the wiki are kept in the [wiki](https://github.com/dungeon-revealer/dungeon-revealer/tree/master/wiki) folder of the [dungeon-revealer repository](https://github.com/dungeon-revealer/dungeon-revealer/).
+To edit the wiki, you can [fork the repository](https://github.com/dungeon-revealer/dungeon-revealer/fork), make changes, and then open a pull request.
+It's the same process as if you were making changes to the code.
+
+
+You can also do all of this on the github website itself.
+Simply navigate to the [wiki](https://github.com/dungeon-revealer/dungeon-revealer/tree/master/wiki) folder, open the file you wish to edit, and click on the edit button.
+Github will take care of the fork and pull request processes for you.
```
Comment (n1ru4l) [inline wiki/Contributing.md:8]: I like that you added this note for non-tech people 👌

**Comment by maxb2** [comment]
> @n1ru4l, can you set a new secret `GH_PAT` with the token generated from [here](https://github.com/settings/tokens/new?scopes=repo&description=wiki%20page%20creator%20token)? It's for [Decathlon/wiki-page-creator-action](https://github.com/marketplace/actions/wiki-page-creator-action)
I can merge once this secret is set. I don't have access to the settings, so I just want to confirm.

**Comment by n1ru4l** [comment]
@maxb2 I added it!

**Pull request merged**